src/dusk.rs

```diff
@@ -213,6 +213,44 @@ impl JubJubExtended {
         let p = JubJubAffine::from(self);
         [p.u, p.v]
     }
+
+    /// Hash an arbitrary slice of bytes to a point on the elliptic curve and
+    /// in the prime order subgroup.
+    ///
+    /// This algorithm uses rejection sampling to hash to a point on the curve:
+    /// The input together with a counter are hashed into an array of 32 bytes.
+    /// If the hash is a canonical representation of a point on the curve and
+    /// a member of the prime-order subgroup, we return it. If not, we increment
+    /// the counter and try to de-serialize again.
+    ///
+    /// **Note:** This implementation of `hash_to_point` is not ideal, in the
+    /// long run we want to implement an algorithm outlined
+    /// [here](https://datatracker.ietf.org/doc/html/rfc9380), but we start with
+    /// this implementation in order to be able to use the API already.
+    pub fn hash_to_point(input: &[u8]) -> Self {
+        let mut counter = 0u64;
+        let mut array = [0u8; 32];
+        loop {
+            let state = blake2b_simd::Params::new()
+                .hash_length(32)
+                .to_state()
+                .update(input)
+                .update(&counter.to_le_bytes())
+                .finalize();
+            let bytes = state.as_bytes();
+
+            array.copy_from_slice(&bytes[..32]);
+
+            if let Ok(point) =
+                <JubJubAffine as Serializable<32>>::from_bytes(&array)
+            {
+                if point.is_prime_order().into() {
```

Comment:
Would it make sense, to add an additional rejection sampling check for if the point is the identity element?

Reply:
Statistically we will never hit the identity element and logically I also don't think that it makes sense. I also wouldn't want that a hash function discards the output `0`.

Reply:
Actually `is_prime_order` *does* check for `!is_identity` so we are already discarding the identity point. I still think that conceptually this doesn't make a lot of sense for a hash function but I'll keep it this way so we don't need to add another check downstream.